### PR TITLE
JIRA# ROS-150: Add printRep, valWrt, setValWrt to Location and create Misc object

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala [hunk 1]
```diff
@@ -59,14 +59,14 @@ object Location {
 
       case LTLexVariable(indirect, level, offset) =>
         k.env.setLex(indirect, level, offset, value) match {
-          case Some(env) => StoreCtxt(k.set(_ >> 'env)(env))
-          case None => StoreFail
+          case Ob.INVALID => StoreFail
```
Comment (tymm) [inline rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: I think the convention is to have the error case second.

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala [hunk 2]
```diff
@@ -59,14 +59,14 @@ object Location {
...
+          case env => StoreCtxt(k.set(_ >> 'env)(env))
         }
 
       case LTAddrVariable(indirect, level, offset) =>
         k.env.setAddr(indirect, level, offset, value) match {
-          case Some(env) => StoreCtxt(k.set(_ >> 'env)(env))
-          case None => StoreFail
+          case Ob.INVALID => StoreFail
```
Comment (tymm) [inline rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: See line 62.

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala [hunk 3]
```diff
@@ -79,8 +79,8 @@ object Location {
       case LTBitField(indirect, level, offset, spanSize, sign) =>
         if (!isFixNum(value)) {
           k.env.setField(indirect, level, offset, spanSize, fixVal(value)) match {
-            case Some(env) => StoreCtxt(k.set(_ >> 'env)(env))
-            case None => StoreFail
+            case Ob.INVALID => StoreFail
```
Comment (tymm) [inline rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: See line 62.

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala [hunk 4]
```diff
@@ -89,8 +89,8 @@ object Location {
       case LTBitField00(offset, spanSize, sign) =>
         if (!isFixNum(value)) {
           k.env.setField(0, 0, offset, spanSize, fixVal(value)) match {
-            case Some(env) => StoreCtxt(k.set(_ >> 'env)(env))
-            case None => StoreFail
+            case Ob.INVALID => StoreFail
```
Comment (tymm) [inline rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: See line 62.

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala [hunk 5]
```diff
@@ -136,4 +136,103 @@ object Location {
...
+                     "code",
+                     "ctxt",
+                     "self",
+                     "self-env",
+                     "rcvr",
+                     "monitor")
+
+    loc.genericType match {
+      case LTCtxtRegister(reg) =>
+        if (0 < reg && reg < Location.NumberOfCtxtRegs) {
```
Comment (tymm) [inline rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: In the C++ implementation it is `0 <= reg`.

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala [hunk 6]
```diff
@@ -136,4 +136,103 @@ object Location {
...
+          names(reg)
+        } else {
+          s"unknown ctxt register 0x$reg%x"
```
Comment (tymm) [inline rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: Why the `%x`?

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala [hunk 7]
```diff
@@ -136,4 +136,103 @@ object Location {
...
+        }
+
+      case LTArgRegister(argReg) =>
+        s"arg[$argReg]"
+
+      case LTLexVariable(indirect, level, offset) => {
```
Comment (tymm) [inline rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala:163]: I think you mixed up lines here? Should be something with `lex[%d,(%d)]`.

File: rosette-scala/src/main/scala/coop/rchain/rosette/Misc.scala [hunk 8]
```diff
@@ -0,0 +1,8 @@
+package coop.rchain.rosette
+
+object Misc {
+  def suicide(msg: String): Unit = {
```
Comment (tymm) [inline rosette-scala/src/main/scala/coop/rchain/rosette/Misc.scala]: Let's put that in the `coop.rchain.rosette` package object? http://www.scala-lang.org/docu/files/packageobjects/packageobjects.html

**Review by tymm — CHANGES_REQUESTED** [review]

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala (hunk 1 above)
Comment (ghost) [reply rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: Done

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala (hunk 2 above)
Comment (ghost) [reply rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: Done

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala (hunk 3 above)
Comment (ghost) [reply rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: Done

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala (hunk 4 above)
Comment (ghost) [reply rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: Done

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala (hunk 5 above)
Comment (ghost) [reply rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: Good catch, thanks!

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala (hunk 6 above)
Comment (ghost) [reply rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala]: Sorry, that should be an f interpolator, not an s interpolator.
https://docs.scala-lang.org/overviews/core/string-interpolation.html#the-f-interpolator

File: rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala (hunk 7 above)
Comment (ghost) [reply rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala:163]: Fixed.

File: rosette-scala/src/main/scala/coop/rchain/rosette/Misc.scala (hunk 8 above)
Comment (ghost) [reply rosette-scala/src/main/scala/coop/rchain/rosette/Misc.scala]: Done

**Comment by ghost** [comment]
Ping

**Review by tymm — APPROVED** [review]

**Comment by tymm** [comment]
Sorry for the delay. I was too focused on my work...

**Pull request merged**